src/emulator/functionsEmulatorShared.ts

```diff
@@ -135,6 +142,65 @@ export class EmulatedTrigger {
   }
 }
 
+/**
+ * Creates a unique trigger definition from Endpoints.
+ * @param Endpoints A list of all CloudFunctions in the deployment.
+ * @return A list of all CloudFunctions in the deployment.
+ */
+export function emulatedFunctionsFromEndpoints(endpoints: Endpoint[]): EmulatedTriggerDefinition[] {
+  const regionDefinitions: EmulatedTriggerDefinition[] = [];
+  for (const endpoint of endpoints) {
+    if (!endpoint.region) {
+      endpoint.region = "us-central1";
+    }
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const def: EmulatedTriggerDefinition = {
+      entryPoint: endpoint.entryPoint,
+      platform: endpoint.platform,
+      region: endpoint.region,
+      // TODO: Difference in use of name/id in Endpoint vs Emulator is subtle and confusing.
+      // We should later refactor the emulator to stop using a custom trigger definition.
+      name: endpoint.id,
+      id: `${endpoint.region}-${endpoint.id}`,
+    };
+    copyIfPresent(def, endpoint, "timeout", "availableMemoryMb", "labels", "platform");
+    // TODO: This transformation is confusing but must be kept since the Firestore/RTDB trigger registration
+    // process requires it in this form. Need to work in Firestore emulator for a proper fix...
+    if (isHttpsTriggered(endpoint)) {
+      def.httpsTrigger = endpoint.httpsTrigger;
+    } else if (isEventTriggered(endpoint)) {
+      const eventTrigger = endpoint.eventTrigger;
+      if (endpoint.platform === "gcfv1") {
+        def.eventTrigger = {
+          eventType: eventTrigger.eventType,
+          resource: eventTrigger.eventFilters.resource,
+        };
+      } else {
+        // Only pubsub and storage events are supported for gcfv2.
+        const { resource, topic, bucket } = endpoint.eventTrigger.eventFilters;
```

Comment:
Clever, but be careful. For example, topic is an id not a name now. Will that get us in trouble? Also, contrary to my original hopes, it seems like "resource" will not be used for v2 stuff.

Reply:
Fortunately, no. 

This code path will need revision as we revise the Endpoint definition in the container contract.